scalyr_agent/agent_main.py

```diff
@@ -794,8 +928,15 @@ def __detailed_status(
         # We wait for five seconds at most to get the status.
         deadline = time.time() + 5
 
+        last_debug_stat_time = 0
         # Now loop until we see it show up.
         while True:
+            # Capture debug stats every second while waiting.
+            if debug and (last_debug_stat_time + 1 < time.time()):
+                stats = self._get_system_and_agent_stats(status_file_path=status_file)
```

Comment:
I wonder if it would make sense to also try to gather output from last_status.tmp file to see if there is a race and something else is also requesting output to be written to that file (temporary file we write status to before we move it in an atomic fashion to a final location).

Reply:
I added creation of the `last_status.tmp` file by using the `mkstemp` as you've suggested in slack, so I guess, it may be enough just so see how many temp files now exist in the directory (by extending existing `ls` command) to see if there are multiple status requests at the same time. WDYT?